### PR TITLE
Refactoring StorageHelper class

### PR DESCRIPTION
**Pull request by jaelim-ms**
* [ ] ~Has `CHANGELOG.md` been updated?~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 1]
```diff
@@ -508,38 +379,28 @@ SQLiteDatabase getDatabase() throws RuntimeException {
...
-     * @param helper A {@link SQLiteOpenHelper} instance to be used for accessing database.
+     * @return An array of column names.
      */
     @VisibleForTesting
-    void setSQLiteOpenHelper(@NonNull SQLiteOpenHelper helper) {
-        mSQLiteOpenHelper.close();
-        mSQLiteOpenHelper = helper;
+    String[] getColumnNames() {
+
+        // TODO: Below line doesn't look efficient to get column names. Could use "PRAGMA table_info(table-name)" to avoid getting all data back from database just to get column names.
```
Comment (dhei) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: How about let's do this TODO now? 😄

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java [hunk 2]
```diff
@@ -71,15 +71,44 @@ public void databaseOperationException() throws Persistence.PersistenceException
...
+        try {
+            assertEquals(0, persistence.countLogs("test-p1"));
+        } finally {
+
+            /* Close. */
+            //noinspection ThrowFromFinallyBlock
+            persistence.close();
+        }
+
+        /* There is a error log. */
```
Comment (dhei) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java]: `a` => `an`

Comment (dhei) [reply sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java]: Please fix all occurrence in this file

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java [hunk 3]
```diff
@@ -78,24 +81,26 @@
     public static void setUpClass() {
         AppCenter.setLogLevel(android.util.Log.VERBOSE);
         sContext = InstrumentationRegistry.getTargetContext();
-        StorageHelper.initialize(sContext);
+        FileManager.initialize(sContext);
+        SharedPreferencesManager.initialize(sContext);
         Constants.loadFromContext(sContext);
     }
 
+    private static int getCursorSize(Cursor cursor) {
```
Comment (guperrot) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/persistence/DatabasePersistenceAndroidTest.java]: Can use cursor.getCount() now that we removed a layer.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 4]
```diff
@@ -232,7 +230,10 @@ public long putLog(@NonNull String group, @NonNull Log log) throws PersistenceEx
                 targetToken = null;
             }
             contentValues = getContentValues(group, isLargePayload ? null : payload, targetToken, log.getType(), targetKey);
-            long databaseId = mDatabaseStorage.put(contentValues);
+            long databaseId = mDatabaseManager.put(contentValues);
+            if (databaseId == -1) {
+                AppCenterLog.warn(LOG_TAG, "Failed to store a log to the Persistence database for log type " + log.getType() + ".");
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: Need to throw PersistenceException

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 5]
```diff
@@ -331,9 +332,16 @@ public void deleteLogs(String group) {
...
-        DatabaseStorage.DatabaseScanner scanner = mDatabaseStorage.getScanner(COLUMN_GROUP, group, null, null, true);
-        int count = scanner.getCount();
-        scanner.close();
+        SQLiteQueryBuilder builder = SQLiteUtils.newSQLiteQueryBuilder();
+        builder.appendWhere(COLUMN_GROUP + " = ?");
+        int count = 0;
+        try {
+            Cursor cursor = mDatabaseManager.getCursor(builder, new String[]{group}, true);
+            count = cursor.getCount();
+            cursor.close();
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: try finally

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 6]
```diff
@@ -473,6 +505,28 @@ public void clearPendingLogState() {
...
+            cursor = mDatabaseManager.getCursor(builder, selectionArgs, true);
+            while (cursor.moveToNext()) {
+                ContentValues idValues = mDatabaseManager.buildValues(cursor);
+                Long invalidId = idValues.getAsLong(DatabaseManager.PRIMARY_KEY);
+                result.add(invalidId);
+            }
+        } catch (RuntimeException e) {
+            AppCenterLog.error(LOG_TAG, "Failed to get corrupted ids: ", e);
+        }
+        if (cursor != null) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: in finally

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 7]
```diff
@@ -317,172 +286,74 @@ public ContentValues get(@IntRange(from = 0) long id) {
...
-                }
-            }
+    public ContentValues get(String key, Object value) {
+        try {
+            SQLiteQueryBuilder builder = SQLiteUtils.newSQLiteQueryBuilder();
+            builder.appendWhere(key + " = ?");
+            String[] selectionArgs = new String[]{value.toString()};
+            Cursor cursor = getCursor(builder, selectionArgs, false);
+            ContentValues values = cursor.moveToFirst() ? buildValues(cursor) : null;
+            cursor.close();
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: finally

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: This is used only in unit tests we could just remove the comment

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java [hunk 8]
```diff
@@ -71,15 +71,44 @@ public void databaseOperationException() throws Persistence.PersistenceException
...
         AppCenterLog.error(eq(AppCenter.LOG_TAG), anyString(), any(RuntimeException.class));
     }
 
     @Test
-    public void clearPendingLogState() throws JSONException {
+    public void clearPendingLogState() throws Exception {
 
         /* groupCount should be <= 9. */
-        final int groupCount = 4;
+        final int groupCount = 1;
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/persistence/DatabasePersistenceTest.java]: what is this change?

**Pull request merged**